actors/market/tests/harness.rs

```diff
@@ -387,10 +413,23 @@ pub fn batch_activate_deals_raw(
     provider: Address,
     sectors_deals: Vec<SectorDeals>,
     compute_cid: bool,
+    expected_activated_deals: Vec<DealID>,
```

Comment:
@anorth I tried hacking this but it's tricky  because the current mockruntime event emission code immediately matches the event being emitted against the latest expected event. See here:

https://github.com/filecoin-project/builtin-actors/blob/d06215ba1e7391a2053684f5bcee20d935711cc1/runtime/src/test_utils.rs#L1261

I tried refactoring this so that all expected events and emitted events are checked together in the end instead being checked during emission but it needs some more refactor in the mock runtime. I think we can punt this for now.

Reply:
Ok, can you please write an issue for that, summarising both our points above, and then link to that from here?